### DISCUSSION TITLE
Potential Bug in k_truss function of core.py

### DISCUSSION THREAD

**Discussion by PurviChaurasia** [networkx/networkx#6500] [Q&A]
In the `k_truss` function in `networkx/algorithms/core.py` it is mentioned that the function doesn't work for graphs containing self loops. However, having tried a few cases on my own behalf, I realised that it is actually giving output for graphs with self loops. Although it creates a copy of the graph using `copy` function therefore any potential self loops are eliminated, this creates a confusion between the two situations. We should either:

- Raise a `NetworkX Error: Self loops not permitted` error in case such an input is given
- Change the documentation and give a possible warning for this scenario

Current Behaviour
<img width="575" alt="image" src="https://user-images.githubusercontent.com/97350598/224690290-c0ac44df-9091-4a03-977f-6898a8a970d2.png">

@MridulS Let me know if I have misunderstood its functionality somewhere :)

**Comment by navyagarwal** [comment]
The k_truss method raises `NetworkXError` when the graph is Directed or a Multigraph.

<img width="673" alt="image" src="https://user-images.githubusercontent.com/82928853/224710043-d0742be5-da65-4cf6-bf85-05dcc0bd75d9.png">

As far as I know, **self loops aren't allowed in undirected graphs**. But the nx.Graph class allows self loops, maybe that's the reason for making a copy that gets rid of self loops in an undirected graph in the method implementation, it's probably like a simpler way of handling things!

> <img width="475" alt="image" src="https://user-images.githubusercontent.com/82928853/224713054-4fb4d2bd-236a-41c1-9930-563e1dcba944.png">

(Sidenote, the output in the image doesn't seem to add up with the input edges)

**Reply by PurviChaurasia** [reply]
Yep, so sorry added the wrong image. Thanks for pointing it out! Here's the correct image.
<img width="325" alt="image" src="https://user-images.githubusercontent.com/97350598/224714370-6dfdb304-5046-486c-8e23-fb98152c0245.png">

And that's exactly my point! I think in that case there needs to be a change in the documentation as it mentions that it raises `NetworkX` error in case of self loops as well.
<img width="442" alt="image" src="https://user-images.githubusercontent.com/97350598/224714461-2905fb80-560b-4e9c-a573-09967ffe5f81.png">
